assets/src/components/AssignmentGradeInput/AssignmentGradeInput.js

```diff
@@ -0,0 +1,75 @@
+import React, { useEffect, useState, useRef } from 'react'
+// import PropTypes from 'prop-types'
+import StyledTextField from '../StyledTextField'
+import { roundToXDecimals, getDecimalPlaceOfFloat } from '../../util/math'
+import debounce from 'lodash.debounce'
+
+function AssignmentGradeInput (props) {
+  const {
+    assignment,
+    className,
+    courseGoalGradeSet,
+    handleAssignmentGoalGrade,
+    handleInputFocus,
+    handleInputBlur,
+    gradeKey
+  } = props
+
+  // Use decimal place of pointsPossible if it's a decimal; otherwise, round to nearest tenth
+  const placeToRoundTo = pointsPossible => (String(pointsPossible).includes('.'))
+    ? getDecimalPlaceOfFloat(pointsPossible) : 1
+
+  const previousGrade = useRef(null)
+  const [goalGradeInternal, setGoalGradeInternal] = useState(roundToXDecimals(assignment.goalGrade, placeToRoundTo(assignment.pointsPossible)))
+  const debouncedGoalGrade = useRef(debounce(grade => {
```

Comment:
I refactored event logging code and the debounced course goal changes for capturing the event state was [simple](https://github.com/tl-its-umich-edu/my-learning-analytics/pull/1173/files#diff-a3e94e4ca1f8654fcea1383709ce1ac2e87c622a28ac43e4635f48b89855813aL31). I am hoping it would be simple of the debounce assignment goal changes as well

Now things are logging only what changed and capturing the previous value as well